lib/galaxy/tools/parameters/validation.py

```diff
@@ -103,20 +121,54 @@ class InRangeValidator( Validator ):
         ...
     ValueError: Not gonna happen
     """
+
     @classmethod
     def from_element( cls, param, elem ):
-        return cls( elem.get( 'message', None ), elem.get( 'min' ), elem.get( 'max' ) )
-    def __init__( self, message, range_min, range_max ):
+        return cls( elem.get( 'message', None ), elem.get( 'min' ),
+                    elem.get( 'max' ), elem.get( 'strict_min', 'false' ), 
+                    elem.get( 'strict_max', 'false' ) )
+
+    def __init__( self, message, range_min, range_max, strict_min=False, strict_max=False ):
+        """
+        When the optional strict_min and strict_max attributes are set
+        to true, the range excludes the end points (i.e., min < value < max),
+        while if set to False ( the default), then range includes the end points
+        (1.e., min <= value <= max).  Combinations of strict_min and strict_max
+        values are allowed.
+        """
         self.min = float( range_min if range_min is not None else '-inf' )
+        self.strict_min = util.asbool( strict_min )
         self.max = float( range_max if range_max is not None else 'inf' )
+        self.strict_max = util.asbool( strict_max )
         assert self.min <= self.max, 'min must be less than or equal to max'
         # Remove unneeded 0s and decimal from floats to make message pretty.
         self_min_str = str( self.min ).rstrip( '0' ).rstrip( '.' )
         self_max_str = str( self.max ).rstrip( '0' ).rstrip( '.' )
-        self.message = message or "Value must be between %s and %s" % ( self_min_str, self_max_str )
+        op1 = '>='
+        op2 = '<='
+        if self.strict_min and self.strict_max:
```

Comment:
@bgruening Maybe I'm not seeing it, but the various checks are needed only so that the one message line can use the defined operators based on the values of strict_min and strict_max.  I'm not seeing how it would be easier to set op1 and op2 to various combinations of <, <=, >, >= and still use a single message line.  Let me know what I'm missing and I'll update the pull request.  Thanks!